x/k6/baggage.go

```diff
@@ -40,11 +47,26 @@ func (lh *labelHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	lh.innerHandler.ServeHTTP(w, r.WithContext(ctx))
 }
 
+func setBaggageContextFromHeader(r *http.Request) (*http.Request, bool) {
+	baggageHeader := r.Header.Get("Baggage")
+	if baggageHeader == "" {
+		return r, false
+	}
+
+	b, err := baggage.Parse(baggageHeader)
+	if err != nil {
+		return r, false
+	}
+
+	ctx := baggage.ContextWithBaggage(r.Context(), b)
+	return r.WithContext(ctx), true
+}
+
```

Comment:
I'm wondering if we always have to do this. I expect that OTel baggage might be already handled in the middleware chain – should we check it with `baggage.FromContext(r.Context())` first?

Reply:
I agree that parsing the baggage in this middleware might be doing too much. 

However, I assume parsing does provide a safeguard by validating and normalizing the baggage(?). What if we:
1. Keep the current implementation
2. Add a comment explaining this reasoning and the assumption of parsed baggage
3. Offer a separate, optional middleware for baggage parsing

Reply:
This method is largely a convenience feature for users so they can get set up with the k6 integration more quickly. I'm not opposed to removing this and requiring users to do this plumbing on their own, but I did think it would make it easier to adopt initially.

Keep in mind this parsing doesn't happen every request. It will only occur when there

- the baggage has not already been parsed and placed in the context and
- there is a `Bagggage` header

If those two conditions are true, then we parse the baggage. Again, happy to pull this out if the consensus is this shouldn't exist in our library code.

Reply:
> the baggage has not already been parsed and placed in the context

From the implementation, it appears that whenever there's a Baggage header present, the `Parse` function is always called, regardless of whether the baggage has been parsed before (maybe I'm missing something).

I also reviewed the OTel baggage `Parse` function implementation, it appears there's no internal mechanism to avoid re-parsing.

Reply:
I believe that this is not an issue, and it should not block you. Just shared my thoughts – something to take into account before we make this GA